pkg/controller/logstorage/logstorage.go

```diff
@@ -161,6 +162,17 @@ func (r *ReconcileLogStorage) createLogStorage(
 		return reconcile.Result{}, false, finalizerCleanup, err
 	}
 
+	// esInternalPublicCertSecret is a secret handled by ECK
```

Comment:
Is that the case with CertificateManagement also? I think the answer should be yes but I'm wondering if you confirmed it with someone that would know for sure.

Reply:
For CertificateManagement the secret I have checked with @rene-dekker that it should have an OwnerReference. Updated the component creation to make it so

Reply:
@rene-dekker could you explain why we should put an OwnerReference on this? Does ECK operator get configured differently where it won't create the `tigera-secure-es-http-certs-public` in the `tigera-elasticsearch` namespace?

Reply:
If we do not create the secret, ECK will create it with the wrong contents. The contents should be installation.spec.certificatemanagement.cacert. When logstorage is removed, the secret should be garbage collected so the namespace can be terminated.

Reply:
A secret will be destroyed when a namespace is deleted so we don't really need to be worried about that.
But that is the first I've heard about ECK creating the wrong cert, we probably need to at a minimum remove the comment I added about it being an optimization on the non CertificateManagement case and replace it with a comment detailing that problem.

Reply:
If no certificate management is used and we do not create the public secret:

   - ECK takes the private secret and creates a public secret with just the tls.crt contents. (For the purpose of other components, like kb to mount)

If no certificate management is used no private secret is created:

   - ECK creates a private secret
   - ECK takes the private secret and creates a public secret with just the tls.crt contents. (For the purpose of other components, like kb to mount)
    
Whereas with cert management we want everything to mount installation.spec.certificatemanagement.cacert.